reactioncommerce/reaction
# "Deny when out of stock" behavior improvements  1928

Resolves #1928 

**Description Part 1**

If I have 5 in stock and I add 10 of something to my cart my quantity is reduced to 5 (which makes sense) but I don't receive any sort of alert that my quantity has been changed and the "Added to Cart" alert shows the original quantity, not the adjusted quantity. This could lead to user confusion. I feel like at one time there was an alert that said "Your quantity has been adjusted".

**Test Steps**
1. Set product variant quantity to 5
2. Add 10 products of the variant onto cart 
3. Expect an alert showing that the original quantity has been adjusted.

**Description Part 2**

If I have 5 in stock and I add 10 to my cart my quantity is reduced to 5. If I add another 20, the total quantity is now 10. So the "Deny when out of stock" is only the affecting the quantity I can add to cart, not the quantity in my cart. So I could have 10 in my cart even though there are only 5 in-stock, which I would think is exactly what "Deny when out of stock" is trying to avoid.

**Test Steps**
1. Set product variant quantity to 5
2. Add 10 products of the variant onto cart 
3. Expect that quantity has been reduced to 5 (An alert pops up for the adjustment)
4. Add another 10 products onto cart
5. Expect an error alert stating that your cart is full
